gzipi/lib.py

```diff
@@ -146,41 +169,139 @@
 _BYTES_IN_KiB = 1024
 
 
-def _is_valid_gzip_header(gzip_header):
+class Compression(enum.Enum):
+    NONE = None
+    GZIP = 'gzip'
+    ZSTD = 'zstd'
+
+
+SUPPORTED_COMPRESSIONS = {Compression.GZIP.value, Compression.ZSTD.value, }
+
+
+def get_compression_format(path: str) -> Optional[str]:
+    """Get compression format based on the extension."""
+    if not isinstance(path, str):
+        return Compression.NONE.value
+    if path.endswith('.gz'):
+        return Compression.GZIP.value
+    elif path.endswith('.zst'):
+        return Compression.ZSTD.value
+    else:
+        return Compression.NONE.value
+
+
+def _get_compression_type(path: str) -> Optional[str]:
+    with open(path, 'rb') as fin:
+        return _get_compression_type_from_fin(fin)
+
+
+def _get_compression_type_from_fin(fin: IO) -> Optional[str]:
```

Comment:
Make it explicit that we're dealing with a binary stream. Also, avoid using ambiguous "get" where possible.

```suggestion
def _determine_compression_from_header(fin: IO[bytes]) -> Optional[str]:
```